cloud-regionsrv-client.spec

```diff
@@ -113,7 +113,7 @@ Requires:       cloud-regionsrv-client >= 6.0.0
 Guest registration plugin for images intended for Google Compute Engine
 
 %package plugin-ec2
-Version:        1.0.4
+Version:        1.0.5
```

Comment:
It's not ok to bump the version in a pull request. Version updates should be done via 

```
bumpversion major|minor|patch
```

at the time when we prepare for a new release such that all commits and fixes between the version and its tags can be simply detected and collected and also compared with the changes entry to be correct and complete. Please revert that part of the PR, it also doesn't seem to be done via bumpversion but rather manually which increases the risk to forget source paths that uses version information

Reply:
Also see `.bumpversion.cfg` file in the sources for a better understanding what a version bump does

Reply:
If the plugin sub-packages should take different versions than the main package I'm ok with your change, if not we should make them bumpversion aware.

Thanks

Reply:
Looking closer to the spec I see that every sub-package has its own version, why is it done that way ? This makes maintenance more complicated in my eyes and also harms the concept of a major version for cloud-regionsrv-client unless the sub-packages are completely independent of the main package, which I believe is not the case

Reply:
The sub packages (plugins) follow their own versioning. The reason behind this is that for the most part (99+%) we do not want to release new plugins when the main implementation changes. This separation reduces churn in the maintenance queue of SLE since code changes to the plugins are very rare.

Reply:
well maybe they are completely independent. If they are we can keep this. Please confirm ?

Reply:
Yes they are independent. A plugin has to provide the `generateRegionSrvArgs()` function. That is the only connection point to the rest of the code base.

Reply:
And a little more detail. The plugin is configured via `regionsrvclnt.cfg` as `instanceArgs` option of the `instance`, as such it is completely optional. The plugin is loaded via `__get_framework_plugin` in `registerutils.py`.

Further evidence that these are independent is that the gce plugin has not changed since it's initial implementation and remains at `1.0.0`. The Azure plugin has only changed once when IMDS support was added in addition to supporting the wireserver protocol. The EC2 plugin had some iterations because of the introduction of local zones and other changes to the data of the `availability-zone` end point. With eth switch to the `region` endpoint we are hopefully done touching that code.

Reply:
ok thanks for clarification